### PR TITLE
Feat: Expand Solana Vault Swap functionality and fix bouncer

### PR DESCRIPTION
**Pull request by albert-llimos**
# Pull Request

Closes: PRO-1697

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [ ] I have written sufficient tests.
- [ ] I have written and tested required migrations.
- [ ] I have updated documentation where appropriate.

## Summary

- Update image with the new SwapEndpoint that has the `creation_slot`
- Add support for Vault Swap Attributes
- Add support for Solana Vault CCM swaps
- Fix bouncer issues all around.
- Properly import new types from the Solana program repo after PRO-1710.

Solana Vault swaps are now working end to end. The SDK doesn't have the vault swap logic but it does have the encoding of CfParameters. For the time being we're just manually doing the swap calls in the bouncer and using some dummy parameters for Vault Swap parameters. When it's moved to the SDK I'll expand the testing to also check for `VaultSwapParameters` and some other small things worth testing.

This code might require some tweaks when PRO-1760 and PRO-1743 are implemented but we'll do that separately.

### CODE REVIEW

**Review by ramizhasan111 — APPROVED** [review]

File: engine/src/witness/sol/program_swaps_witnessing.rs [hunk 1]
```diff
@@ -290,7 +334,8 @@ async fn get_program_swap_event_accounts_data(
 				Err(anyhow!("Expected UiAccountData::Binary(String, UiAccountEncoding::Base64)")),
 			None => Ok((account, None)),
```
Comment (ramizhasan111) [inline engine/src/witness/sol/program_swaps_witnessing.rs:335]: why this change?

**Review by ramizhasan111 — APPROVED** [review]

File: engine/src/witness/sol/program_swaps_witnessing.rs [hunk 2]
```diff
@@ -290,7 +334,8 @@ async fn get_program_swap_event_accounts_data(
...
 		})
-		.collect()
+		.collect();
+	program_swap_event_accounts_iter
```
Comment (ramizhasan111) [inline engine/src/witness/sol/program_swaps_witnessing.rs:338]: why this change?

Comment (albert-llimos) [reply engine/src/witness/sol/program_swaps_witnessing.rs:338]: oh, it's just because I had added some printing before. Will revert.
